source/common/network/transport_socket_options_impl.cc

```diff
@@ -16,5 +18,33 @@ void TransportSocketOptionsImpl::hashKey(std::vector<uint8_t>& key) const {
     }
   }
 }
+
+TransportSocketOptionsSharedPtr
+TransportSocketOptionsUtility::fromFilterState(const StreamInfo::FilterState& filter_state) {
```

Comment:
perf nit: WDYT about checking to see if filter_state has any data at all and if there is none (the super common case) just return nullptr immediately? You might need to add an empty method to filter state?

Reply:
The underlying impl is `absl::flat_hash_map<std::string, ObjectPtr>`, I wonder how much faster that one `empty` call is, compared to two `find` call to an empty hash_map. The key is static so we're not allocating any thing. I agree if we do more lookup it might worth, but not today, IMO.

Reply:
OK sounds good.